### PR TITLE
1.0.0 - Plugin Launch

### LINKED ISSUE #2: Error on plugin activation
The "verify_plugin_dependencies" function is not detecting the dependencies properly causing the plugin not begin activated even if it has all dependencies.

- [x] Simplify verification method of existence of the the LifterLMS plugin.

### PR DESCRIPTION
**Pull request by FabricioQueirozLkn**
## Changelog

Lançamento do plugin.

### v1.0.0 - 25/07/23

- Plugin launch.

## Conferir
- [x] Atualizar versão hardcoded
- [x] Atualizar README com notas
- [x] Atualizar CHANGELOG

### CODE REVIEW

File: README.txt [hunk 1]
```diff
@@ -1,114 +1,84 @@
...
+**Dependencies**
 
-*   "Contributors" is a comma separated list of wp.org/wp-plugins.org usernames
-*   "Tags" is a comma separated list of tags that apply to the plugin
-*   "Requires at least" is the lowest version that the plugin will work on
-*   "Tested up to" is the highest version that you've *successfully used to test the plugin*. Note that it might work on
-higher versions... this is just the highest one you've verified.
-*   Stable tag should indicate the Subversion "tag" of the latest stable version, or "trunk," if you use `/trunk/` for
-stable.
+Payment Banking Slip Pix for LifterLMS plugin is dependent on LifterLMS plugin, please make sure LifterLMS is installed and properly configured before starting Payment Banking Slip Pix for LifterLMS installation.
```
Comment (emanuellopess) [inline README.txt]: @FabricioLinkNacional Could you put a link to the Lifter LMS plugin? In the "[..] on LifterLMS plugin, [..]" put a markdown link to here: https://wordpress.org/plugins/lifterlms/

File: admin/index.php
Comment (emanuellopess) [inline admin/index.php:1]: @FabricioLinkNacional This should follow the pattern "silence is golden", like the example below:
```php
<?php

// Silence is golden
```

File: admin/lkn-payment-banking-slip-pix-for-lifterlms-slip-settings.php [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+
+            // Field for set number of days to due date.
+            $fields[] = array(
+                'id' => $gateway->get_option_name( 'days_due_date' ),
+                'title' => __( 'Days to due date', 'payment-banking-slip-pix-for-lifterlms' ),
+                'desc' => '<br>' . __( 'Defines the number of days until the Bank Slip expiration.', 'payment-banking-slip-pix-for-lifterlms' ),
+                'type' => 'number',
+                'default' => 1,
+                'step' => 1,
+				'min' => 0,
```
Comment (emanuellopess) [inline admin/lkn-payment-banking-slip-pix-for-lifterlms-slip-settings.php]: @FabricioLinkNacional Please verify you linting extension, you need to format this file

File: includes/class-lkn-payment-banking-slip-pix-for-lifterlms-helper.php [hunk 3]
```diff
@@ -127,4 +126,71 @@ final public static function dependency_alert(): void {
...
+            $page_url .= 's';
+        }
+        $page_url .= '://';
+        if ('80' !== $_SERVER['SERVER_PORT']) {
+            $page_url .= $_SERVER['SERVER_NAME'] . ':' . $_SERVER['SERVER_PORT'] . $_SERVER['REQUEST_URI'];
+        } else {
+            $page_url .= $_SERVER['SERVER_NAME'] . $_SERVER['REQUEST_URI'];
+        }
+
+        return $page_url;
```
Comment (emanuellopess) [inline includes/class-lkn-payment-banking-slip-pix-for-lifterlms-helper.php]: @FabricioLinkNacional Needs sanitization, please use [this function](https://developer.wordpress.org/reference/functions/sanitize_url/) to sanitize the $page_url

File: includes/class-lkn-payment-banking-slip-pix-for-lifterlms.php [hunk 4]
```diff
@@ -120,22 +120,50 @@ public function get_version() {
...
-     * Add the PagHiper Payment gateway to the list of available gateways.
-     *
-     * @param array
-     * @param mixed $gateways
+     * Add the PagHiper Payment gateways to the list of available gateways.
+     * 
+     * @since 1.0.0
      */
-    public static function add_gateways($gateways) {
+    public static function add_gateways() {
```
Comment (emanuellopess) [inline includes/class-lkn-payment-banking-slip-pix-for-lifterlms.php]: @FabricioLinkNacional This function needs the parameter $gateways to work, this function is called in a filter, it doesn't have args but it has a parameter, see more [here](https://developer.wordpress.org/reference/functions/apply_filters/)

It actually deletes all gateways except for the Banking Slip and PIX.

File: public/js/lkn-payment-banking-slip-pix-for-lifterlms-public.js [hunk 5]
```diff
@@ -1,31 +1,64 @@
...
+        .catch(function (error) {
+          buttonTitle.tooltip('dispose')
+          buttonTitle.tooltip({
+            title: 'Falha ao copiar código: ', error
+          }).tooltip('show')
+        })
+    })
+  })
+
+  // $(window).load(function () {
```
Comment (emanuellopess) [inline public/js/lkn-payment-banking-slip-pix-for-lifterlms-public.js]: @FabricioLinkNacional Please delete all unused code

File: public/class-lkn-payment-banking-slip-pix-for-lifterlms-public.php [hunk 6]
```diff
@@ -86,6 +86,6 @@ public function enqueue_scripts(): void {
          * class.
          */
 
-        wp_enqueue_script( $this->plugin_name, plugin_dir_url( __FILE__ ) . 'js/lkn-payment-banking-slip-pix-for-lifterlms-public.js', array('jquery'), $this->version, false );
+        wp_enqueue_script( 'lkn-payment-banking-slip-pix-for-lifterlms', plugin_dir_url( __FILE__ ) . 'js/lkn-payment-banking-slip-pix-for-lifterlms-public.js', array('jquery'), $this->version, false );
```
Comment (emanuellopess) [inline public/class-lkn-payment-banking-slip-pix-for-lifterlms-public.php]: @FabricioLinkNacional Add the function '[localize_script](https://developer.wordpress.org/reference/functions/wp_localize_script/)' to load the string translations

File: public/js/lkn-payment-banking-slip-pix-for-lifterlms-public.js [hunk 7]
```diff
@@ -1,31 +1,64 @@
...
+
+    $('#lkn_copy_code').click(function () {
+      const codigoEmv = $('#lkn_emvcode').val()
+      const buttonTitle = $('#lkn_copy_code')
+
+      navigator.clipboard.writeText(codigoEmv)
+        .then(function () {
+          buttonTitle.tooltip('dispose')
+          buttonTitle.tooltip({
+            title: 'Código copiado'
```
Comment (emanuellopess) [inline public/js/lkn-payment-banking-slip-pix-for-lifterlms-public.js]: @FabricioLinkNacional Needs translation

File: public/js/lkn-payment-banking-slip-pix-for-lifterlms-public.js [hunk 8]
```diff
@@ -1,31 +1,64 @@
...
+          }).tooltip('show')
+        })
+        .catch(function (error) {
+          buttonTitle.tooltip('dispose')
+          buttonTitle.tooltip({
+            title: 'Falha ao copiar código: ', error
```
Comment (emanuellopess) [inline public/js/lkn-payment-banking-slip-pix-for-lifterlms-public.js]: @FabricioLinkNacional Needs translation

File: admin/class-lkn-payment-banking-slip-pix-for-lifterlms-admin.php [hunk 9]
```diff
@@ -86,6 +86,18 @@ public function enqueue_scripts(): void {
...
+            'message' => sprintf(
+                __('Get new features with %1$sPayment Banking Slip Pix for LifterLMS Pro.%2$s', 'payment-banking-slip-pix-for-lifterlms' ),
+                '<a href="https://www.linknacional.com/wordpress/plugins/" target="_blank">',
+                '</a>'
+            )
+        );
+
+        wp_register_script('freemium-banner-strings', '');
+
+        wp_localize_script('freemium-banner-strings', 'bannerStrings', $bannerStrings);
```
Comment (emanuellopess) [inline admin/class-lkn-payment-banking-slip-pix-for-lifterlms-admin.php]: @FabricioLinkNacional This is not right, when you are localizing a script you need to register the localization to the script handler, in this case, $this->plugin_name, you are using a unnecessary empty script to load the localization.

**Review by emanuellopess — CHANGES_REQUESTED** [review]
Needs little changes before the WordPress approval, call me if you need more details

File: admin/lkn-payment-banking-slip-pix-for-lifterlms-slip-settings.php (hunk 2 above)
Comment (FabricioQueirozLkn) [reply admin/lkn-payment-banking-slip-pix-for-lifterlms-slip-settings.php]: @emanuellopess It doesn't show up broken like that in my vs code, but i moved it anyway.

File: .vscode/.php-cs-fixer.php [hunk 10]
```diff
@@ -1,20 +1,11 @@
...
-/**
- * php-cs-fixer config that follows the WordPress style guide for PHP.
- * Source: https://gist.github.com/srbrunoferreira/5b0d96955c3913f6b1cd805c2a14079d
- * Adapted from: https://github.com/vena/php-cs-fixer-wordpress/blob/main/src/WordPressRuleSet.php
- * WordPress style guide: https://developer.wordpress.org/coding-standards/wordpress-coding-standards/php/.
- */
-
 return (new PhpCsFixer\Config())
     ->setRules(array(
-        // Built-ins
```
Comment (emanuellopess) [inline .vscode/.php-cs-fixer.php:12]: @FabricioLinkNacional Please view the new linting rules defined in the DEV Normatives

**Review by emanuellopess — APPROVED** [review]

**Comment by emanuellopess** [comment]
@FabricioLinkNacional Adicionar biblioteca de plugin-updater e ajustar o script do workflow

File: .github/workflows/main.yml [hunk 11]
```diff
@@ -0,0 +1,81 @@
...
+
+    # Add plugin files to a root directory
+    - name: Make plugin root directory with auto-updater
+      run: "mkdir ${{env.PLUGIN_NAME_WITH_UPDATER}} && cp -r ./admin ./includes ./languages ./public *.php LICENSE composer.json composer.lock ./${{env.PLUGIN_NAME_WITH_UPDATER}}/ && find ./${{env.PLUGIN_NAME_WITH_UPDATER}} -type f -exec chmod 0644 {} + && find ./${{env.PLUGIN_NAME_WITH_UPDATER}} -type d -exec chmod 0755 {} + && ls -lah"
+
+    - name: Make plugin root directory
+      run: "rm -Rf ./includes/plugin-updater && mkdir ${{env.PLUGIN_NAME}} && mv -t ./${{env.PLUGIN_NAME}} ./admin ./includes ./languages ./public *.php LICENSE composer.json composer.lock *.txt && find ./${{env.PLUGIN_NAME}} -type f -exec chmod 0644 {} + && find ./${{env.PLUGIN_NAME}} -type d -exec chmod 0755 {} + && ls -lah"
+
+    # Run composer install
+    - name: Run composer install
```
Comment (ghost) [inline .github/workflows/main.yml:22]: composer install deve rodar antes do passo "Make plugin root directory with auto-updater".

E no passo "Make plugin root directory with auto-updater", deve ser incluída a pasta vendor.

File: CHANGELOG.md
Comment (ghost) [inline CHANGELOG.md:1]: Mesmo sendo a primeira versão, acho interessante inserir quais features o plugin carrega na sua primeira versão.

File: .github/workflows/main.yml (hunk 11 above)
Comment (FabricioQueirozLkn) [reply .github/workflows/main.yml:22]: @brunoferreiralkn  saquei, então nem preciso fazer o composer install duas vezes né, mas no caso ele vai incluido nos dois comandos de "Make plugin root directory" ? Ou só no com Updater ?

**Review by ghost — CHANGES_REQUESTED** [review]
Falta implementar o workflow corretamente.

File: .github/workflows/main.yml (hunk 11 above)
Comment (ghost) [reply .github/workflows/main.yml:22]: Vai nas duas build e o composer install só precisa rodar uma vez.

**Review by ghost — APPROVED** [review]

**Pull request merged**